pointonsoftware/pscore
# Use the latest Ubuntu version in builds

xxxxx

- Use the latest ubuntu version in the PR workflow
- Update GCC version to 10
- Update cpp check suppression
- Address cppcheck issues
- Update findVectorContent using stl algorithm

### Pull Request Checklist

-   [x] I have ensured that my commit message contains the Issue ID.
-   [x] I have ensured that my commit message contains the feature/bugfix description.
-   [x] I have ensured my code follows the coding guidelines.
-   [x] I have ran lint in my code locally prior to submission.
-   [x] I have built my changes in local successfully.
